Fix/bugs in movie details screen

monaayman123 opened the pull request:
# Pull Request Template

## Description
This PR fixes existing UI/logic bugs in the movie and TV show details screens, improves error handling, and refactors the code to improve readability and reusability.
The main purpose is to provide a smoother user experience by fixing issues (,loading states not resetting, padding issues) and breaking down large composable blocks into smaller, reusable functions.
---

## Changes Made
Bug Fixes:

Fixed issue where movie/TV show posters were missing in non-English languages by adding a poster fallback from the English version.

Fixed "Add to List" button remaining in loading state when reopening the dialog after a failed attempt.

Fixed movie details app bar having excessive top padding.

Fixed release date not formatting according to the selected app language.

Refactoring:

Extracted movie poster UI into a MoviePosterItem extension function.

Extracted movie extras section rendering into a separate MovieExtrasSectionContent function.

Extracted main movie details sections (title, genres, info, description, cast, extras) into their own LazyListScope extension functions for reusability and better readability.

Improved PlayButton shadow visibility by adjusting blur and alpha values.

Refactor localization utilities
- Added `toLocalizedNumbers` for Arabic numeral conversion.
- Added `getLocalizedTimeUnit` for localized time units (h/m).
- Added `reverseDateFormat` for reversing date strings.
- Added `formatMovieLength` to convert movie length to localized hours and minutes.
- Added `localizeCountryCode` to display country names in Arabic or codes in English.
- Consolidated all localization-related helper functions into `LocalizationUtils.kt`.

UI Improvements:

Reduced app bar padding for better visual balance.

Improved spacing and alignment in movie details screen.

---
### Localization Refactor

Removed LocalizationUtils.kt from the ui module.
Deleted MovieInfoSection.kt and replaced its usage with DottedSeparatedRow.
Updated MovieDetailsUiState to include currentLanguage.
Modified MovieDetailsViewModel to fetch and store the current language.
Adjusted MovieDetailsUiStateMapper to use the currentLanguage for localizing movie details.
Updated MovieDetailsScreen.kt and MoreLikeSection.kt to reflect the new data structure and remove direct calls to deleted localization utilities.

Duration State Refactor

- Moved DurationUiState to the shared module to avoid duplication.
- Updated SeriesDetailsViewModel to use non-nullable tvShowId from args.
- Utilized toLocalizedName mapper for origin country in SeriesDetailsStateMapper.

Bug Fix

- Fixed issue where the country was displayed in English short code (e.g., "US", "FR") even when the app language was Arabic.
- Now shows full Arabic country names when language = Arabic.
- Otherwise falls back to English short code.

## Screenshots (if applicable)

<img width="363" height="285" alt="Screenshot_20250815_033542" src="https://github.com/user-attachments/assets/0f4d9b93-ce09-4a41-bec1-8b0255963dbc" />
<img width="406" height="274" alt="Screenshot_20250815_034043" src="https://github.com/user-attachments/assets/91a7e90f-fa08-4e9a-91b7-2ced79c57825" />
<img width="463" height="332" alt="Screenshot_20250815_035040" src="https://github.com/user-attachments/assets/03c156a5-451e-4c3e-bb4c-9172e2b12e7c" />
<img width="1344" height="2030" alt="Screenshot_20250815_051658" src="https://github.com/user-attachments/assets/07d89213-b074-4397-a6fd-fb8dce967fdc" />
<img width="1344" height="1311" alt="Screenshot_20250815_133628" src="https://github.com/user-attachments/assets/b943efac-0027-4a6e-b44e-69938a15f470" />
<img width="1344" height="176" alt="Screenshot_20250815_131036" src="https://github.com/user-attachments/assets/0d3811f2-9e21-47dd-9f68-76625446b2a0" />

---
### Localization Screenshot
Arabic:
<img width="231" height="135" alt="{B12C8FE7-8E48-4D44-8413-E5765CAC2188}" src="https://github.com/user-attachments/assets/b864927b-a581-4f5b-a9dd-9598793f39f3" />

English:
<img width="267" height="101" alt="{6ADDABFF-3648-4334-A158-12E7969F528D}" src="https://github.com/user-attachments/assets/113f5f98-5f33-4c07-aa5a-38c48157a2bf" />

---

## Checklist
Please ensure the following tasks are completed:

- [ ] My code follows the code style of this project
- [ ] Changes have been tested manually and verified.
- [ ] PR includes at most one single feature.

---

## Additional Comments
Add any additional information or context about the pull request here.

monaayman123 committed:
fix: reset isAddMovieToListLoading on error to prevent stuck loading state after dialog reopen

monaayman123 committed:
fix: add @ sign before username in review card

monaayman123 committed:
fix: remove padding from bottom in AddToListDialog.kt dialog

monaayman123 committed:
fix: remove padding from top in app
bar

monaayman123 committed:
refactor: add playButtonShadow color in design system

monaayman123 committed:
refactor: add playButtonBlur in active state and remove it in disable state

monaayman123 committed:
refactor: extract movie details UI sections into separate  functions for better readability and reusability

monaayman123 committed:
refactor:  localization utilities:

Added localizeCountryCode function to display full country names in Arabic and country codes in English.

Updated MovieInfoSection to use localized country names.

Ensures numbers and time units are properly localized based on current language.

monaayman123 committed:
refactor: remove @ sign

monaayman123 committed:
refactor: rename file from FormateDate.kt to be LocalizationUtils.kt

Belal-Alkhatib commented on ui/src/main/java/com/amsterdam/ui/utils/LocalizationUtils.kt:
```diff
@@ -0,0 +1,90 @@
+        year = parts[0]
+    }
+    return "$year-$month-$day".toLocalizedNumbers()
+}
```
Since we already have a mapper for converting domain classes to UI state classes, this date formatting logic should be moved there to keep transformation responsibilities in one place and maintain cleaner separation of concerns

Belal-Alkhatib commented on ui/src/main/java/com/amsterdam/ui/utils/LocalizationUtils.kt:
```diff
@@ -0,0 +1,90 @@
+        "ar" -> arMap[key] ?: unit
+        else -> enMap[key] ?: unit
+    }
+}
```
Instead of manually mapping time unit strings here, we could create a composable function that takes the length (either as an integer or a class containing hours and minutes) and returns a localized formatted string by using the numeric values with string resources. This approach would scale better for multiple languages and keep translations consistent across the app.

ahmedmahgoub0 commented:
drop shadow of play button not same as design

monaayman123 committed:
Merge branch 'develop' of https://github.com/Amsterdam-Team/Aflami into fix/bugs-in-movie-details-screen

# Conflicts:
#	designSystem/src/main/java/com/amsterdam/designsystem/theme/colors/DarkColors.kt
#	designSystem/src/main/java/com/amsterdam/designsystem/theme/colors/LigthColors.kt
#	ui/src/main/java/com/amsterdam/ui/screens/movieDetails/MovieDetailsScreen.kt
#	viewModel/src/main/java/com/amsterdam/viewmodel/movieDetails/MovieDetailsViewModel.kt

monaayman123 committed:
refactor: resolve conflicts

monaayman123 committed:
Merge branch 'develop' of https://github.com/Amsterdam-Team/Aflami into fix/bugs-in-movie-details-screen

Belal-Alkhatib committed:
refactor: move `DurationUiState` to shared module and add mapper

Belal-Alkhatib committed:
refactor: Remove localization utilities from UI layer

This commit removes the `LocalizationUtils.kt` file from the `ui` module as localization concerns are now handled in the `viewModel` layer.

The following changes were made:
- Deleted `LocalizationUtils.kt`.
- Deleted `MovieInfoSection.kt` and replaced its usage with `DottedSeparatedRow`.
- Updated `MovieDetailsUiState` to include `currentLanguage`.
- Modified `MovieDetailsViewModel` to fetch and store the current language.
- Adjusted `MovieDetailsUiStateMapper` to use the `currentLanguage` for localizing movie details.
- Updated `MovieDetailsScreen.kt` and `moreLikeSection.kt` to reflect the new data structure and remove direct calls to deleted localization utilities.

Belal-Alkhatib committed:
refactor: move DurationUiState to shared module

- Moved `DurationUiState` to the shared module to avoid code duplication.
- Updated `SeriesDetailsViewModel` to use non-nullable `tvShowId` from args.
- Utilized `toLocalizedName` mapper for origin country in `SeriesDetailsStateMapper`.

monaayman123 committed:
Merge branch 'develop' of https://github.com/Amsterdam-Team/Aflami into fix/bugs-in-movie-details-screen

# Conflicts:
#	ui/src/main/java/com/amsterdam/ui/components/movieAndTvShowDetails/PlayButton.kt
#	ui/src/main/java/com/amsterdam/ui/screens/movieDetails/MovieDetailsScreen.kt

monaayman123 committed:
Merge remote-tracking branch 'origin/fix/bugs-in-movie-details-screen' into fix/bugs-in-movie-details-screen

mohannadahmed00 commented on viewModel/src/main/java/com/amsterdam/viewmodel/movieDetails/MovieDetailsViewModel.kt:
```diff
@@ -49,7 +49,8 @@ class MovieDetailsViewModel @Inject constructor(
 
         manageLocaleLanguageUseCase.getAppLanguage()
-            .onEach {
+            .onEach { language ->
```
what will happen if the flow emits exception???
